src/tests/ftest/util/server_utils_params.py

```diff
@@ -319,202 +314,455 @@ def update_log_files(self, control_log, helper_log, server_log):
+        self.pinned_numa_node = BasicParameter(None)
+        self.log_mask = BasicParameter(None, "INFO")
+        self.log_file = LogParameter(log_dir, None, "daos_server.log")
+
+        # Set default environment variables
+        default_env_vars = [
+            "ABT_ENV_MAX_NUM_XSTREAMS=100",
+            "ABT_MAX_NUM_XSTREAMS=100",
+            "DAOS_MD_CAP=1024",
+            "DAOS_SCHED_WATCHDOG_ALL=1",
+            "DD_MASK=mgmt,io,md,epc,rebuild",
+        ]
+        default_env_vars.extend(self.REQUIRED_ENV_VARS["common"])
+        for name in self._provider.split(";"):
+            if name in self.REQUIRED_ENV_VARS:
+                default_env_vars.extend(self.REQUIRED_ENV_VARS[name])
+        self.env_vars = BasicParameter(None, default_env_vars)
+
+        # global CRT_CTX_SHARE_ADDR shared with client
+        self.crt_ctx_share_addr = BasicParameter(None, default_share_addr)
+
+        # the storage configuration for this engine
+        self.storage = StorageYamlParameters(self._index, max_storage_tiers)
+
+    def get_params(self, test):
+        """Get values for the daos server yaml config file.
+
+        Args:
+            test (Test): avocado Test object
+        """
+        super().get_params(test)
+
+        # Override the log file file name with the test log file name
+        if hasattr(test, "server_log") and test.server_log is not None:
+            self.log_file.value = test.server_log
+
+        # Create the requested number of storage tier parameters
+        self.storage.get_params(test)
+
+        # Define any required env vars
+        required_env_vars = {}
+        for env in self.REQUIRED_ENV_VARS["common"]:
+            required_env_vars[env.split("=", maxsplit=1)[0]] = env.split("=", maxsplit=1)[1]
+        for name in self._provider.split(";"):
+            if name in self.REQUIRED_ENV_VARS:
+                required_env_vars.update({
+                    env.split("=", maxsplit=1)[0]: env.split("=", maxsplit=1)[1]
+                    for env in self.REQUIRED_ENV_VARS[name]})
+
+        # Enable fault injection if configured
+        if test.fault_injection.fault_file is not None:
+            self.log.debug("Enabling fault injection")
+            required_env_vars["D_FI_CONFIG"] = test.fault_injection.fault_file
+
+        # Update the env vars with any missing or different required setting
+        update = False
+        env_var_dict = {env.split("=")[0]: env.split("=")[1] for env in self.env_vars.value}
+        for key in sorted(required_env_vars):
+            if key not in env_var_dict or env_var_dict[key] != required_env_vars[key]:
+                env_var_dict[key] = required_env_vars[key]
+                update = True
+        if update:
+            self.log.debug("Assigning required env_vars")
+            new_env_vars = ["=".join([key, str(value)]) for key, value in env_var_dict.items()]
+            self.env_vars.update(new_env_vars, "env_var")
```

Comment:
Would `EnvironmentVariables.from_list` or `EnvironmentVariables.update_from_list` be helpful here?

Reply:
Possibly, but with everything else going on in this PR I wasn't looking to change this code.